### PR TITLE
{bio}[goolf-1.4.10] PyMOL and {devel}[goolf-1.4.10] GLEW (REVIEW)

### PR DESCRIPTION
**Pull request by a-mazur**
PyMOL is an important app for Structural Biology. It would be nice to have it in EB.
There is a newer version 1.7.4.0 but it does not compile correctly

### CODE REVIEW

**Comment by a-mazur** [comment]
Test report by @a-mazur
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux centos 6.5, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/82c84134eb81e4e14977 for a full test report.

File: easybuild/easyconfigs/g/GLEW/GLEW-1.10.0-goolf-1.4.10.eb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+sources = [SOURCELOWER_TGZ]
+
+# this is not standard " configure && make && make install" so let's tweak it
+skipsteps = ['configure', 'install']
+
+# we setup as env vars the system to compile and the install dir
+premakeopts = 'export SYSTEM=linux && export GLEW_DEST=%(installdir)s && '
+
+# and we do "make all && make install.all"
+makeopts = 'all && make install.all'
```
Comment (boegel) [inline easybuild/easyconfigs/g/GLEW/GLEW-1.10.0-goolf-1.4.10.eb]: this should be `buildopts`, Jenkins will not like this (`makeopts` is no longer supported in EasyBuild v2.0)

File: easybuild/easyconfigs/g/GLEW/GLEW-1.10.0-goolf-1.4.10.eb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+toolchain = {'name': 'goolf', 'version': '1.4.10'}
+
+source_urls = [SOURCEFORGE_SOURCE]
+sources = [SOURCELOWER_TGZ]
+
+# this is not standard " configure && make && make install" so let's tweak it
+skipsteps = ['configure', 'install']
+
+# we setup as env vars the system to compile and the install dir
+premakeopts = 'export SYSTEM=linux && export GLEW_DEST=%(installdir)s && '
```
Comment (boegel) [inline easybuild/easyconfigs/g/GLEW/GLEW-1.10.0-goolf-1.4.10.eb]: `prebuildopts`

Comment (boegel) [inline easybuild/easyconfigs/g/GLEW/GLEW-1.10.0-goolf-1.4.10.eb]: there's no need to export I think, see if this works:

```python
prebuildopts = "SYSTEM=linux GLEW_DEST=%(installdir)s"
```

File: easybuild/easyconfigs/g/GLEW/GLEW-1.10.0-goolf-1.4.10.eb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+# and we do "make all && make install.all"
+makeopts = 'all && make install.all'
+
+osdependencies = ['libXi-devel', 'mesa-libGL-devel', 'freeglut-devel']
```
Comment (boegel) [inline easybuild/easyconfigs/g/GLEW/GLEW-1.10.0-goolf-1.4.10.eb]: we have EasyBuild dependencies for each of these, but that's quite a bag of worms...

not sure about this

File: easybuild/easyconfigs/p/PyMOL/PyMOL-1.7.2.1-goolf-1.4.10-Python-2.7.5.eb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+versionsuffix = "-%s-%s" % (python, pythonversion)
+
+dependencies = [
+    (python, pythonversion),
+    ('Pmw', '1.3.3', '-%s-%s' % (python, pythonversion)),
+    ('GLEW', '1.10.0'),
+]
+
+osdependencies = ['libXi-devel', 'mesa-libGL-devel', 'freeglut-devel']
```
Comment (boegel) [inline easybuild/easyconfigs/p/PyMOL/PyMOL-1.7.2.1-goolf-1.4.10-Python-2.7.5.eb]: same here, use EB deps?

but `Mesa` pulls in a huge pile of crap, see #1226

File: easybuild/easyconfigs/p/PyMOL/PyMOL-1.7.2.1-goolf-1.4.10-Python-2.7.5.eb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+description = """ PyMOL is a powerful and comprehensive molecular visualization 
+ product for rendering and animating 3D molecular structures"""
+
+toolchain = {'name': 'goolf', 'version': '1.4.10'}
+
+source_urls = [SOURCEFORGE_SOURCE]
+sources = ['%(namelower)s-v%(version)s.tar.bz2']
+
+python = "Python"
+pythonversion = '2.7.5'
```
Comment (boegel) [inline easybuild/easyconfigs/p/PyMOL/PyMOL-1.7.2.1-goolf-1.4.10-Python-2.7.5.eb]: bump to `2.7.9`?

File: easybuild/easyconfigs/p/PyMOL/PyMOL-1.7.2.1-goolf-1.4.10-Python-2.7.5.eb [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+pythonshortversion = ".".join(pythonversion.split(".")[:-1])
```
Comment (boegel) [inline easybuild/easyconfigs/p/PyMOL/PyMOL-1.7.2.1-goolf-1.4.10-Python-2.7.5.eb]: I prefer `'.'` over `"."`, but I'm really nitpicking now

**Comment by a-mazur** [comment]
Test report by @a-mazur
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux centos 6.5, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/8dc202e76577d4afb9ea for a full test report.

File: easybuild/easyconfigs/g/GLEW/GLEW-1.10.0-goolf-1.4.10.eb [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+# we setup as env vars the system to compile and the install dir
+prebuildopts = 'SYSTEM=linux && export GLEW_DEST=%(installdir)s && '
+
+# and we do "make all && make install.all"
+buildopts = 'all && make install.all'
+
+python = 'Python'
+pythonversion = '2.7.5'
+
```
Comment (boegel) [inline easybuild/easyconfigs/g/GLEW/GLEW-1.10.0-goolf-1.4.10.eb]: include a `versionsuffix` based on this:

```python
versionsuffix = '-%s-%s' % (python, pythonversion)
```

File: easybuild/easyconfigs/g/GLEW/GLEW-1.10.0-goolf-1.4.10.eb [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+dependencies = [
+    ('freeglut', '2.8.1'),
+    ('libXi', '1.7.2'),
+    ('Mesa', '7.11.2', '-%s-%s' % (python, pythonversion)),
```
Comment (boegel) [inline easybuild/easyconfigs/g/GLEW/GLEW-1.10.0-goolf-1.4.10.eb]: replace the last bit by `versionsuffix`?

**Comment by boegel** [comment]
@a-mazur: I would be in favor of replacing the `Mesa` easyconfig you include here with the one being contributed in #1226, since that one doesn't include any OS dependencies anymore...

Thoughts?

If we don't mind holding this one until #1226 is ready and merged, we can update this PR when #1226 is merged (no use in doing it earlier, since @1226 is not ready yet).

**Comment by a-mazur** [comment]
@boegel: I have no problem with waiting for clean Mesa easyconfig. Is it fine otherwise?

**Comment by boegel** [comment]
@a-mazur: looks fine otherwise, yeah, let me trigger Jenkins too and then follow-up once #1226 makes it in (maybe state that you're waiting for that one in that PR)

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
Jenkins: test this please

File: easybuild/easyconfigs/g/GLEW/GLEW-1.10.0-goolf-1.4.10.eb [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+source_urls = [SOURCEFORGE_SOURCE]
+sources = [SOURCELOWER_TGZ]
+
+# this is not standard " configure && make && make install" so let's tweak it
+skipsteps = ['configure', 'install']
+
+# we setup as env vars the system to compile and the install dir
+prebuildopts = 'SYSTEM=linux && export GLEW_DEST=%(installdir)s && '
+
+# and we do "make all && make install.all"
```
Comment (fgeorgatos) [inline easybuild/easyconfigs/g/GLEW/GLEW-1.10.0-goolf-1.4.10.eb]: L27 redundant

File: easybuild/easyconfigs/l/libxml2/libxml2-2.9.1-goolf-1.4.10-Python-2.7.5.eb [hunk 10]
```diff
@@ -0,0 +1,26 @@
+name = 'libxml2'
+version = '2.9.1'
+
+homepage = 'http://xmlsoft.org/'
+description = """Libxml2 is the XML C parser and toolchain developed for the Gnome project (but usable
+outside of the Gnome platform)."""
+toolchain = {'name': 'goolf', 'version': '1.4.10'}
```
Comment (fgeorgatos) [inline easybuild/easyconfigs/l/libxml2/libxml2-2.9.1-goolf-1.4.10-Python-2.7.5.eb]: help above 3 lines in readability... (split descr., empty line etc)

File: easybuild/easyconfigs/m/Mesa/Mesa-7.11.2-goolf-1.4.10-Python-2.7.5.eb [hunk 11]
```diff
@@ -0,0 +1,56 @@
+easyblock = 'ConfigureMake'
+
+name = 'Mesa'
+version = '7.11.2'
+
+homepage = 'http://www.mesa3d.org/'
+description = """Mesa is an open-source implementation of the OpenGL specification -
+a system for rendering interactive 3D graphics."""
+toolchain = {'name': 'goolf', 'version': '1.4.10'}
+toolchainopts = {'optarch': True}
```
Comment (fgeorgatos) [inline easybuild/easyconfigs/m/Mesa/Mesa-7.11.2-goolf-1.4.10-Python-2.7.5.eb]: readability issues, like above...

File: easybuild/easyconfigs/m/Mesa/Mesa-7.11.2-goolf-1.4.10-Python-2.7.5.eb [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+        ('glproto', '1.4.16'),
+        ('libdrm', '2.4.27')
+        ]
+
+osdependencies = [
+            'libX11-devel',  # Xlibs.h
+            'xorg-x11-proto-devel',  # X.h, glproto, xproto
+            'libXdamage-devel', 
+            'libXext-devel', 
+            'libXfixes-devel',
```
Comment (fgeorgatos) [inline easybuild/easyconfigs/m/Mesa/Mesa-7.11.2-goolf-1.4.10-Python-2.7.5.eb]: 4-space indents above, no indent below for the square bracket

**Comment by fgeorgatos** [comment]
@a-mazur: this is largely ok, just finish up the last touches

**Comment by migueldiascosta** [comment]
@a-mazur, this PR is being closed for the following reason(s): no activity for > 6 months.
Please don't hesitate to reopen this PR or add a comment if you feel this contribution is still relevant.
For more information on our policy w.r.t. closing PRs, see https://easybuild.readthedocs.io/en/latest/Contributing.html#why-a-pull-request-may-be-closed-by-a-maintainer

**Pull request closed without merging**